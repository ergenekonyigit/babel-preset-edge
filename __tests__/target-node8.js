import buildPreset from "../src"
import { fixtures, titles, check } from "./core"

describe("Node v8", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    target: "node8"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
