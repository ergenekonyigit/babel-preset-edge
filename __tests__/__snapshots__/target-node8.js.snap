// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node v8 comments 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

function translate(value) {
  return _camelCase(value);
}"
`;

exports[`Node v8 deadcode 1`] = `"const STATIC = false;"`;

exports[`Node v8 es3: literal 1`] = `"foo.catch;"`;

exports[`Node v8 es3: member 1`] = `
"var foo = {
  catch: function () {}
};"
`;

exports[`Node v8 es2015: classes 1`] = `
"export class Component {
  main() {}

}"
`;

exports[`Node v8 es2015: for_off 1`] = `
"function test() {
  for (const taskName of tasks) {
    try {
      executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Node v8 es2015: generator 1`] = `
"function* idMaker() {
  var index = 0;

  while (true) yield index++;
}"
`;

exports[`Node v8 es2015: set 1`] = `"new Set([1, 2, 3]);"`;

exports[`Node v8 es2016: exponentiation 1`] = `"var result = 8;"`;

exports[`Node v8 esnext: async 1`] = `
"async function wait() {
  await loadData();
}"
`;

exports[`Node v8 esnext: async_loop 1`] = `
"async function executeTasks(tasks) {
  for (const taskName of tasks) {
    try {
      await executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Node v8 esnext: class properties 1`] = `
"import _defineProperty from \\"@babel/runtime/helpers/builtin/es6/defineProperty\\";
export class Component {
  constructor() {
    _defineProperty(this, \\"onClick\\", () => {});
  }

}"
`;

exports[`Node v8 esnext: dynamic import 1`] = `
"Promise.resolve().then(() => require(\\"lodash\\")).then(lodash => {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`Node v8 esnext: import 1`] = `
"import _camelCase from \\"lodash/camelCase\\";
console.log(_camelCase(\\"hello world\\"));"
`;

exports[`Node v8 esnext: object spread 1`] = `
"import _objectSpread from \\"@babel/runtime/helpers/builtin/es6/objectSpread\\";

let original = {
  bar: 10
},
    variant = _objectSpread({
  foo: 1
}, original);"
`;

exports[`Node v8 lodash: async 1`] = `
"import _parallel from \\"async/parallel\\";

_parallel([]);"
`;

exports[`Node v8 lodash: lodash 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

_camelCase(\\"hello world\\");"
`;

exports[`Node v8 lodash: recompose 1`] = `
"import _compose from \\"recompose/compose\\";

_compose([]);"
`;

exports[`Node v8 react: jsx 1`] = `
"function render() {
  return React.createElement(\\"h1\\", {
    __self: this
  }, \\"Hello\\");
}"
`;

exports[`Node v8 react: jsx spread 1`] = `
"function render(props) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, props, {
    __self: this
  }), \\"Hello\\");
}"
`;

exports[`Node v8 react: proptypes 1`] = `
"import React from \\"react\\";
import PropTypes from \\"prop-types\\";

class MyComponent extends React.Component {
  render() {
    return React.createElement(\\"div\\", {
      __self: this
    }, this.props.userName);
  }

}

MyComponent.propTypes = {
  userName: PropTypes.string
};"
`;
